presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayDistinctFromOperator.java

```diff
@@ -47,11 +47,8 @@ public static boolean isDistinctFrom(
             @SqlType("array(E)") Block right,
             @IsNull boolean rightNull)
     {
-        if (leftNull != rightNull) {
-            return true;
-        }
-        if (leftNull) {
-            return false;
+        if (leftNull || rightNull) {
+            return leftNull ^ rightNull;
```

Comment:
I believe `!=` is more proper here than `^` because we are not trying to do a bitwise operation here. `^` is essentially the bitwise counterpart of `!=`. Ditto for the other occurrences.